src/lockscreen.js

```diff
@@ -42,6 +49,39 @@ var Lockscreen = GObject.registerClass({
         this.notify('locked');
     }
 
+    get key() {
+        return this._key;
+    }
+
+    set key(key) {
+        if ('_key' in this && this._key === key)
+            return;
+        this._key = key;
+        this._updateLockState();
+    }
+
+    _onChanged() {
+        this._updateLockState();
+
+        /* TODO check for key and display can-be-opened graphics */
+    }
+
+    _updateLockState() {
+        if (!this._key)
+            return;
+        if (this._manager.isUnlocked(this._key))
+            this.locked = false;
+    }
+
+    _onClicked() {
+        if (!this._key)
+            return;
+        if (!this._manager.hasKey(this._key))
+            return;
+        this.locked = false;
+        this._manager.setUnlocked(this._key);
```

Comment:
Wouldn't it be more practical to have the `locked` setter take care of also calling `setUnlocked()`?

Reply:
it's a bit confusing but, there is only 1 specific scenario where we should set it to "unlocked" in the GameStateService, and that is when the player has used the key.

In other scenarios, e.g. when the locked has been unlocked in a previous session, then setting the lockscreen.locked to false should only affect UI, not the GameStateService.